src/Microsoft.OData.Client/ALinq/ResourceBinder.cs

```diff
@@ -46,6 +44,11 @@ internal class ResourceBinder : DataServiceALinqExpressionVisitor
         /// </summary>
         private readonly DataServiceContext context;
 
+        /// <summary>
+        /// If true, a Where clause that compares only the key property, will generate a $filter query option.
+        /// </summary>
+        private static bool keyComparisonGeneratesFilterQuery;

```

Comment:
Defining a static property on a non-static seems sketchy. Maybe just add context as a parameter to the AnalyzePredicate method and remove the property on the binder?

Reply:
#Fixed